test/modules/generic-mqtt-tester/src/tester.rs

```diff
@@ -44,18 +47,46 @@ impl MessageTesterShutdownHandle {
         }
     }
 
-    pub async fn shutdown(mut self) -> Result<(), MessageTesterError> {
-        self.poll_client_shutdown
-            .send(())
-            .await
-            .map_err(MessageTesterError::SendShutdownSignal)?;
-        self.message_channel_shutdown.shutdown().await?;
+    pub async fn shutdown(mut self, exited: ExitedWork) {
+        match exited {
+            ExitedWork::NoneOrUnknown => {
+                self.shutdown_message_initiator().await;
+                self.shutdown_message_channel().await;
+                self.shutdown_poll_client().await;
+            }
+            ExitedWork::MessageChannel => {
+                self.shutdown_message_initiator().await;
+                self.shutdown_poll_client().await;
+            }
+            ExitedWork::MessageInitiator => {
+                self.shutdown_message_channel().await;
+                self.shutdown_poll_client().await;
+            }
+            ExitedWork::PollClient => {
+                self.shutdown_message_initiator().await;
+                self.shutdown_message_channel().await;
+            }
+        }
+    }
```

Comment:
I appreciate your efforts to make shutdown process better, but after all, this is a test module, not a product. Maybe we could just do our best and fire and forget shutdown signals and disregard errors if we try to double shutdown? It is just a suggestion, thought. I'm fine with what we have here.

Reply:
I agree, just wanted to prove to myself I could solve the issue. Didn't take too long so thought it was worth the learning exercise.